pytest_fixtures/satellite_auth.py

```diff
@@ -0,0 +1,167 @@
+        groups_base=ldap_data['group_base_dn'],
+        attr_firstname=LDAP_ATTR['firstname'],
+        attr_lastname=LDAP_ATTR['surname'],
+        attr_login=LDAP_ATTR['login_ad'],
+        server_type=LDAP_SERVER_TYPE['API']['ad'],
+        attr_mail=LDAP_ATTR['mail'],
+        name=gen_string('alpha'),
+        host=ldap_data['ldap_hostname'],
+        tls=False,
+        port='389',
+        organization=[module_org],
+        location=[module_loc],
+    ).create()
+
+
+@fixture(scope='function')
+def auth_source_ipa(ipa_data, module_org, module_loc):
+    return entities.AuthSourceLDAP(
+        onthefly_register=True,
+        account=ipa_data['ldap_ipa_user_name'],
+        account_password=ipa_data['ldap_ipa_user_passwd'],
+        base_dn=ipa_data['ipa_base_dn'],
+        groups_base=ipa_data['ipa_group_base_dn'],
+        attr_firstname=LDAP_ATTR['firstname'],
+        attr_lastname=LDAP_ATTR['surname'],
+        attr_login=LDAP_ATTR['login'],
+        server_type=LDAP_SERVER_TYPE['API']['ipa'],
+        attr_mail=LDAP_ATTR['mail'],
+        name=gen_string('alpha'),
+        host=ipa_data['ldap_ipa_hostname'],
+        tls=False,
+        port='389',
+        organization=[module_org],
+        location=[module_loc],
+    ).create()
+
+
+@fixture()
+def ldap_user_name(ldap_data, test_name):
+    """Add LDAP user to satellite by logging in, return username to test and delete the user (if
+    still exists) when test finishes.
+    """
+    with Session(test_name, ldap_data['ldap_user_name'], ldap_data['ldap_user_passwd']):
+        pass
+    yield ldap_data['ldap_user_name']
+    users = entities.User().search(
+        query={'search': 'login="{}"'.format(ldap_data['ldap_user_name'])}
+    )
+    if users:
+        users[0].delete()
+
+
+@fixture()
+def ldap_usergroup_name():
+    """Return some random usergroup name, and attempt to delete such usergroup when test finishes.
+    """
+    usergroup_name = gen_string('alphanumeric')
+    yield usergroup_name
+    user_groups = entities.UserGroup().search(query={'search': 'name="{}"'.format(usergroup_name)})
+    if user_groups:
+        user_groups[0].delete()
+
+
+@fixture(scope='module')
+def ldap_auth_name():
+    """Return some random ldap name, and attempt to delete all ldap when test starts.
+    """
+    ldap = entities.AuthSourceLDAP().search()
+    for ldap_auth in range(len(ldap)):
+        users = entities.User(auth_source=ldap[ldap_auth]).search()
+        for user in range(len(users)):
+            users[user].delete()
+        ldap[ldap_auth].delete()
+    ldap_name = gen_string('alphanumeric')
+    yield ldap_name
+
+
+@fixture(scope='module')
+def enroll_idm_and_configure_external_auth():
+    """Enroll the Satellite6 Server to an IDM Server."""
+    run_command(cmd='yum -y --disableplugin=foreman-protector install ipa-client ipa-admintools')
+
+    run_command(
+        cmd='echo {0} | kinit admin'.format(settings.ipa.password_ipa),
```

Comment:
Replace `.format()` with f-string formatting.
```suggestion
        cmd=f'echo {settings.ipa.password_ipa} | kinit admin',
```

Reply:
Do this for all over in your PR.